File: 1/Tree.java
import java.util.LinkedList;
import java.util.Queue;

public class Tree {
	private Node root = null;	

	public class Node {
		public int value;
		public Node left = null;
		public Node right = null;

		Node(int value) {
			this.value = value;
		}
	}

	public Node getRoot() {
		return root;
	}

	public void insert(int value) {
		
		if(root == null) {
			root = new Node(value);
			return;
		} 
		
		Queue<Node> q = new LinkedList<Node>();
		q.add(root);
		
		while(!q.isEmpty()) {
			Node temp = q.remove();

			if(temp.left == null) {
				temp.left = new Node(value);
				break;
			} else {
					q.add(temp.left);
			}

			if(temp.right == null) {
				temp.right = new Node(value);
				break;
			} else {
					q.add(temp.right);
			}
		}

	}
	
	public void insert(int values[]) {
		for(int value : values) {
			this.insert(value);
		}
	}

}

File: 1/js/src/Tree.ts
class TreeNode {
	data: number;
	left: TreeNode | null;
	right: TreeNode | null;

	constructor(data: number) {
		this.data = data;
		this.left = this.right = null;
	}
}

class Tree {
	root: TreeNode | null;
	
	constructor() {
		this.root = null;
	}

	insertOne(data: number) {
		const newNode = new TreeNode(data);
		
		if(this.root == null) {
			this.root = newNode;
			return;
		}

		let queue: TreeNode[] = [];
		queue.push(this.root);

		while(queue.length != 0) {
			const currentNode: TreeNode = queue.shift()!;
			
			if(currentNode.left == null) {
				currentNode.left = newNode;
				return;
			} else {
				queue.push(currentNode.left);
			}
			
			if(currentNode.right == null) {
				currentNode.right = newNode;
				return;
			} else {
				queue.push(currentNode.right);
			}
		}
	}

	insertMany(dataArr: number[]) {
		for(const data of dataArr) {
			this.insertOne(data);
		}
	}
}

export {TreeNode, Tree};
